packages/batch-submitter/src/transaction-chain-contract.ts

```diff
@@ -38,17 +62,21 @@ const appendSequencerBatch = async (
   batch: AppendSequencerBatchParams,
   options?: TransactionRequest
 ): Promise<TransactionResponse> => {
-  const methodId = keccak256(
-    Buffer.from(APPEND_SEQUENCER_BATCH_METHOD_ID)
-  ).slice(2, 10)
-  const calldata = encodeAppendSequencerBatch(batch)
   return OVM_CanonicalTransactionChain.signer.sendTransaction({
     to: OVM_CanonicalTransactionChain.address,
-    data: '0x' + methodId + calldata,
+    data: getEncodedCalldata(batch),
     ...options,
   })
 }
 
+const getEncodedCalldata = (batch: AppendSequencerBatchParams): string => {
+  const methodId = keccak256(
+    Buffer.from(APPEND_SEQUENCER_BATCH_METHOD_ID)
+  ).slice(2, 10)
```

Comment:
Not the hugest deal but the `methodId` could be computed once at startup and held in a global variable